src/frontend/Semantic_check.ml

```diff
@@ -755,6 +747,16 @@ and semantic_check_expression_of_int_or_real_type cf e name =
       Semantic_error.int_or_real_expected ue.emeta.loc name ue.emeta.type_
       |> error)
 
+let semantic_check_expression_of_scalar_or_type cf t e name =
```

Comment:
Minor thing, not sure the or is necessary. I guess a replace all error? :)

Reply:
The "or" _is_ necessary because it's not "scalar type", it's "_either_ scalar _or_ the given type".

I don't like the `scalar_or_type` name but I'm not sure what is a good alternative.

Reply:
Ah, makes sense.